externs/shaka/player.js

```diff
@@ -387,6 +389,9 @@ shaka.extern.TrackList;
  *   The minimum bandwidth of a variant track, in bit/sec.
  * @property {number} maxBandwidth
  *   The maximum bandwidth of a variant track, in bit/sec.
+ * @property {number} maxDisabledTime
```

Comment:
I'm not sure if this is the best site, because the restrictions can be hard or soft, I think it's better to get this out to the `streaming` level.

Reply:
Problem is that `abrManager_.configure` passes only `this.config_.abr`, see:
https://github.com/shaka-project/shaka-player/blob/main/lib/player.js#L1905
What is your suggestion? Passing also `this.config_.streaming` and storing streaming configuration from abrManager?

Reply:
@theodab do you have any suggestions?

Reply:
So, as-is, this is leaving the option up to the developer as to if they want the variant-disabling to disable the track in the track selection. I can't see much use for developers to allow users to continue to select the broken variant in the track selection, so it does make sense for this to be moved outside of the `restrictions` structure.

An alternate approach would be, instead of having a `disabledTime` variable that stores the START of the disabled time period, have a `disabledUntilTime` variable that stores the END of the period. So when it's first set, you'd do:
```js
variant.disabledUntilTime = Date.now() / 1000 + this.config_.streaming.maxDisabledTime;
```
That way, only the player needs to know what `maxDisabledTime` is.

Reply:
Ok, applied suggestion